src/main/java/frc/robot/poseestimator/IOdometryPose.java

```diff
@@ -0,0 +1,20 @@
+package frc.robot.poseestimator;
+
+import edu.wpi.first.math.Matrix;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveDriveWheelPositions;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.numbers.N3;
+
+public interface IOdometryPose {
+
+    void updateOdometry(OdometryObservation odometryObservation);
+
+    void resetOdometry(Rotation2d gyroAngle, SwerveDriveWheelPositions wheelPositions, Pose2d pose);
```

Comment:
Switch that get wheels before rotation, like in the observation and in wpilib estimator functions